blocktx/client.go

```diff
@@ -102,7 +103,7 @@ func (btc *Client) Start(minedBlockChan chan *blocktx_api.Block) {
 						block, err = stream.Recv()
 						if err != nil {
 							btc.logger.Error("Failed to receive block", slog.String("err", err.Error()))
-							break
+							break getBlocksLoop
```

Comment:
could you please document what is actually goingg on here. 
1. Why there are 2 selects?
2. Why do you have to use labels ? Why can't you just continue the loop instead?

Reply:
Each time the `blocktx` service stops there are these logs in `metamorph`

```
...
{"time":"2023-11-24T16:37:18.176813526Z","level":"ERROR","msg":"Failed to receive block","service":"btx","err":"rpc error: code = Unavailable desc = error reading from server: EOF"}
{"time":"2023-11-24T16:37:18.176820856Z","level":"ERROR","msg":"Failed to receive block","service":"btx","err":"rpc error: code = Unavailable desc = error reading from server: EOF"}
{"time":"2023-11-24T16:37:18.176847157Z","level":"ERROR","msg":"Failed to receive block","service":"btx","err":"rpc error: code = Unavailable desc = error reading from server: EOF"}
{"time":"2023-11-24T16:37:18.176855267Z","level":"ERROR","msg":"Failed to receive block","service":"btx","err":"rpc error: code = Unavailable desc = error reading from server: EOF"}
{"time":"2023-11-24T16:37:18.176863218Z","level":"ERROR","msg":"Failed to receive block","service":"btx","err":"rpc error: code = Unavailable desc = error reading from server: EOF"}
{"time":"2023-11-24T16:37:18.176870278Z","level":"ERROR","msg":"Failed to receive block","service":"btx","err":"rpc error: code = Unavailable desc = error reading from server: EOF"}
{"time":"2023-11-24T16:37:18.176877387Z","level":"ERROR","msg":"Failed to receive block","service":"btx","err":"rpc error: code = Unavailable desc = error reading from server: EOF"}
{"time":"2023-11-24T16:37:18.176884637Z","level":"ERROR","msg":"Failed to receive block","service":"btx","err":"rpc error: code = Unavailable desc = error reading from server: EOF"}
{"time":"2023-11-24T16:37:18.176892287Z","level":"ERROR","msg":"Failed to receive block","service":"btx","err":"rpc error: code = Unavailable desc = error reading from server: EOF"}
{"time":"2023-11-24T16:37:18.176913888Z","level":"ERROR","msg":"Failed to receive 
...
```

So the client has this infinite loop of trying to receive a block instead of waiting and creating a new connection. 
But the label is at the wrong place and I should probably add a test. So I'll close the PR.